jdaviz/configs/default/aida.py

```diff
@@ -0,0 +1,170 @@
+        self.app = viewer.jdaviz_app
+
+    def _get_image_glue_data(self, image_label):
+        if image_label is None:
+            i_top = get_top_layer_index(self.viewer)
+            image = self.viewer.layers[i_top].layer
+
+        else:
+            for lyr in self.viewer.layers:
+                image = lyr.layer
+                if image.label == image_label:
+                    break
+            else:
+                raise ValueError(f"No data with data_label {image_label}` found in viewer.")
+
+        return image, image.label
+
+    def set_viewport(self, center=None, fov=None, image_label=None, **kwargs):
+        """
+        Parameters
+        ----------
+        center : `~astropy.coordinates.SkyCoord` or tuple of floats
+            Center the viewer on this coordinate.
+
+        fov : `~astropy.units.Quantity` or tuple of floats
+            Set the width of the viewport to span `field_of_view`.
+
+        image_label : str
+            Set the viewport with respect to the image
+            with the data label: ``image_label``.
+        """
+        image, image_label = self._get_image_glue_data(image_label)
+
+        if center is None:
+            # get the current center in the pixel coords on reference data
+            x_min, x_max, y_min, y_max = self.viewer.get_limits()
+            center_x = 0.5 * (x_min + x_max)
+            center_y = 0.5 * (y_min + y_max)
+            center = (center_x, center_y)
+
+        if isinstance(center, SkyCoord):
+            reference_wcs = self.viewer.state.reference_data.coords
+
+            if isinstance(reference_wcs, GWCS):
+                reference_wcs = WCS(reference_wcs.to_fits_sip())
+
+            reference_center_pix = reference_wcs.world_to_pixel(center)
+
+        elif hasattr(center, '__len__') and isinstance(center[0], (float, int)):
+            reference_center_pix = center
+
+        current_width = self.viewer.state.x_max - self.viewer.state.x_min
+        current_height = self.viewer.state.y_max - self.viewer.state.y_min
+
+        if fov is None:
+            new_width = current_width
+            new_height = current_height
+        else:
+            if isinstance(fov, (u.Quantity, Angle)):
+                current_fov = self._get_current_fov('sky')
+                scale_factor = float(fov / current_fov)
+
+            elif isinstance(fov, (float, int)):
+                current_fov = self._get_current_fov('pixel')
+                scale_factor = float(fov / current_fov)
+
+            new_width = current_width * scale_factor
+            new_height = current_height * scale_factor
+
+        new_xmin = reference_center_pix[0] - (new_width * 0.5)
+        new_ymin = reference_center_pix[1] - (new_height * 0.5)
+
+        self.viewer.set_limits(
+            x_min=new_xmin,
+            x_max=new_xmin + new_width,
+            y_min=new_ymin,
+            y_max=new_ymin + new_height
+        )
+
+    def _get_current_fov(self, sky_or_pixel):
```

Comment:
nit: if we separate this into multiple function (i.e. `_get_current_fov_sky` and `_get_current_fov_pixel`) we will be more able to support future types as we continue to develop this class. I don't think we'll be returning other types, but its always good to design our services with that in mind

that said, this follows the AIDA convention, so its not the biggest issue